src/libraries/System.Formats.Tar/src/System/Formats/Tar/PosixTarEntry.cs

```diff
@@ -10,16 +10,40 @@ namespace System.Formats.Tar
     /// Even though the <see cref="TarEntryFormat.Gnu"/> format is not POSIX compatible, it implements and supports the Unix-specific fields that were defined in that POSIX standard.</remarks>
     public abstract partial class PosixTarEntry : TarEntry
     {
-        // Constructor used when reading an existing archive.
-        internal PosixTarEntry(TarHeader header, TarReader readerOfOrigin)
-            : base(header, readerOfOrigin)
+        // Constructor called when reading a TarEntry from a TarReader.
+        internal PosixTarEntry(TarHeader header, TarReader readerOfOrigin, TarEntryFormat format)
+            : base(header, readerOfOrigin, format)
         {
         }
 
-        // Constructor called when creating a new 'TarEntry*' instance that can be passed to a TarWriter.
         internal PosixTarEntry(TarEntryType entryType, string entryName, TarEntryFormat format)
             : base(entryType, entryName, format)
         {
+            _header._uName = string.Empty;
+            _header._gName = string.Empty;
+            _header._devMajor = 0;
+            _header._devMinor = 0;
+        }
+
+        // Constructor called when creating a new TarEntry.
+        internal PosixTarEntry(TarEntry other, TarEntryFormat format)
+            : base(other, format)
+        {
+            // Avoid transferring if entry was V7
```

Comment:
I'm not sure what `transferring` means here... and the code doesn't seem to do anything special for V7.

Right before hitting the "add comment" button I think I figured it out, and V7 is the only format that isn't a PosixTarEntry, so the comment is really about the else, not the if...

I'd just delete the comment.  The code speaks for itself, PosixTarEntry entries copy over the POSIX fields, non-Posix entry types (which might be only V7?) don't.

Reply:
Removing comment.